Automattic/wp-calypso
# composite-checkout: Allow removing coupons from cart

#### Changes proposed in this Pull Request

This adds a trash button next to coupon items in the review order step in composite checkout. It is hooked up to the cart manager so that coupons can be removed from the cart, if desired.

![coupon-delete-button](https://user-images.githubusercontent.com/2036909/75831183-dccf6d80-5d80-11ea-8b8f-2a91a25fed5b.png)

#### To do

- [x] When deleting a coupon immediately after adding it, the coupon field still contains the coupon code and also displays an error as though the coupon failed to add. I'm not sure what's causing this.
- [x] The copy in the confirmation dialog is not very applicable to coupons. Maybe we should change it? It currently reads: "You are about to remove your product from the cart; When you press Continue, we will remove your product from the cart and your site will continue to run without it." Honestly that wording is a little confusing anyway. What if we changed it to: "You are about to remove a product from your cart; Press Continue if you are sure you want to remove it."
- [x] If you remove a coupon and you're on the last step, the payment method gets set to the first available one but with invalid data. We should instead switch back to the first step.

#### Testing instructions

Add a coupon to your cart (in either version of checkout), then, using composite checkout, click to remove it from your cart in the review step. Make sure that after you confirm the dialog, the coupon is removed.
